src/main/java/edu/kit/ipd/crowdcontrol/objectservice/database/operations/NotificationOperations.java

```diff
@@ -122,60 +137,80 @@ public boolean delteNotification(int notificationID) {
      * @return {@code true} if deleted, {@code false} otherwise
      */
     public boolean deleteNotification(int id) {
-        NotificationRecord record = create.newRecord(NOTIFICATION);
-        record.setIdNotification(id);
-
-        return create.executeDelete(record, NOTIFICATION.ID_NOTIFICATION.eq(id)) == 1;
+        create.deleteFrom(NOTIFICATION_RECEIVER_EMAIL)
+                .where(NOTIFICATION_RECEIVER_EMAIL.NOTIFICATION.eq(id));
+        return create.deleteFrom(NOTIFICATION).where(NOTIFICATION.ID_NOTIFICATION.eq(id)).execute() == 1;
     }
 
-    /**
-     * Returns a single notification.
-     *
-     * @param id the ID of the notification
-     * @return the notification of empty if not found
-     */
-    public Optional<Notification> getNotification(int id) {
-        return create.fetchOptional(NOTIFICATION, NOTIFICATION.ID_NOTIFICATION.eq(id))
-                .map(NotificationTransformer::toProto);
-    }
-
-    /**
-     * updates the notifications lastSend field
-     * @param notificationID the primary key of the notification
-     * @return tre if updated, false if not found
-     */
-    public boolean updateLastSendForNotification(int notificationID) {
-        //TODO? wait for simon!
-        return true;//create.update(Tables.NOTIFICATION)
-                //.set(Tables.NOTIFICATION.LASTSENT, Timestamp.valueOf(LocalDateTime.now()))
-                //.where(Tables.NOTIFICATION.ID_NOTIFICATION.eq(notificationID))
-                //.execute() == 1;
-    }
 
     /**
      * Updates a notification.
      *
-     * @param id ID of the notification
+     * @param id           ID of the notification
      * @param notification new notification contents
      * @return the updated notification
      */
     public Notification updateNotification(int id, Notification notification) {
-        NotificationRecord record = create
+        NotificationRecord notificationRecord = create
                 .fetchOptional(NOTIFICATION, NOTIFICATION.ID_NOTIFICATION.eq(id))
                 .orElseThrow(() -> new NotFoundException("Notification does not exist!"));
 
-        record = NotificationTransformer.mergeRecord(record, notification);
-        record.update();
+        notificationRecord = NotificationTransformer.mergeRecord(notificationRecord, notification);
+        notificationRecord.update();
+
+        notification.getEmailsList().stream()
+                .map(s -> new NotificationReceiverEmailRecord(null, notification.getId(), s))
+                .collect(Collectors.collectingAndThen(Collectors.toList(), create::batchUpdate))
+                .execute();
```

Comment:
this won't work. update needs a primary key and you never delete existing. (i would do the following: first transform then in NotificationReceiverEmailRecords. then create a **transaction** that deletes the associated and inserts new ones)